system/lib/pthread/library_pthread.c

```diff
@@ -268,7 +268,19 @@ int usleep(unsigned usec)
 	}
 #ifdef __EMSCRIPTEN__
 	emscripten_conditional_set_current_thread_status(EM_THREAD_STATUS_SLEEPING, EM_THREAD_STATUS_RUNNING);
-#endif
+#endif	
+}
+
+int nanosleep(const struct timespec *req, struct timespec *rem)
+{
+	if (!req || req->tv_nsec < 0 || req->tv_nsec > 999999999L || req->tv_sec < 0) return EINVAL;
+	do_sleep(req->tv_sec * 1000.0 + req->tv_nsec / 1e6);
```

Comment:
Oh, when the exponent notation `e` is used, the literal is always a double (unless there is a `f` suffix), so `1e6` is a double. For example,

	#include <stdio.h>

	int main()
	{
		double x = 100 + 10 / 1e6;
		printf("%f\n", x);
	}

will print `100.000010`.